go-controller/pkg/ovn/ovn.go

```diff
@@ -501,13 +502,19 @@ func (oc *Controller) syncNodeGateway(node *kapi.Node, subnet *net.IPNet) error
 
 // WatchNodes starts the watching of node resource and calls
 // back the appropriate handler logic
-func (oc *Controller) WatchNodes(nodeSelector *metav1.LabelSelector) error {
+func (oc *Controller) WatchNodes() error {
 	var gatewaysFailed sync.Map
 	macAddressFailed := make(map[string]bool)
-	_, err := oc.watchFactory.AddFilteredNodeHandler(nodeSelector, cache.ResourceEventHandlerFuncs{
+	nodeSelector, _ := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+		MatchLabels: map[string]string{kapi.LabelOSStable: "linux"},
+	})
+	_, err := oc.watchFactory.AddNodeHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
@JacobTanenbaum now that we're not using a filtered node handler, we need to care about what happens when an ignored node gets an Update event. You could create a new function for `isNodeIgnored(node *kapi.Node)` that checks oc.logicalSwitchCache (don't forget to lock before reading it) that you could use in both addNode() and the Update event handler.